commons/commons-serialization/src/test/kotlin/ac/uk/ebi/biostd/tsv/FileListTsvStreamDeserializerTest.kt

```diff
@@ -55,16 +55,26 @@ class FileListTsvStreamDeserializerTest(
 
     @Test
     fun `serialize - deserialize FileList`() {
-        val files = (1..20_000).map {
+        fun bioFile(it: Int) =
             BioFile("folder/file$it.txt", attributes = listOf(Attribute("Attr1", "A$it"), Attribute("Attr2", "B$it")))
-        }.asSequence()
-        val iterator = files.iterator()
+
+        var idx = -1
+        val files = object : Sequence<BioFile> {
+            override fun iterator(): Iterator<BioFile> = object : Iterator<BioFile> {
```

Comment:
this way of creating a sequence looks overcomplex, why we have change previos approach (create a list and use asSequence).  You may change https://kotlinlang.org/docs/sequences.html#construct there they mention 3 recomended way to construct a sequence. 

1. From elements
2. From iterable (previos appraoch)
3. From a function
4. From chunks﻿

All this IMO looks way simpler that new approach.